tests/Integration/Behaviour/Features/Context/Domain/CarrierFeatureContext.php

```diff
@@ -0,0 +1,482 @@
+
+        if ($isModule !== $actualResult) {
+            throw new RuntimeException(sprintf(
+                'Carrier "%s" shipping price is calculated by %s, but it was expected it to be calculated by %s',
+                $reference,
+                $actualResult ? 'module' : 'PrestaShop',
+                $calculationMethod
+            ));
+        }
+    }
+
+    /**
+     * @Then /^Carrier "(.+)" module (should|should not) need the shipping price calculated by PrestaShop$/
+     */
+    public function assertModuleNeedsCoreShippingPrice($reference, $condition)
+    {
+        /** @var Carrier $carrier */
+        $carrier = SharedStorage::getStorage()->get($reference);
+        $needsPrice = 'should' === $condition;
+        $actualResult = (bool) $carrier->need_range;
+
+        if ($needsPrice !== $actualResult) {
+            throw new RuntimeException(sprintf(
+                'Carrier "%s" module %s the shipping price calculated by PrestaShop, but was expected it to %s',
+                $reference,
+                $actualResult ? 'needs' : 'does not need',
+                $condition === 'should' ? 'need' : 'not need'
+            ));
+        }
+    }
+
+    /**
+     * @Then /^Carrier "(.+)" (should|should not) belong to module$/
+     */
+    public function assertIsModule($reference, $condition)
+    {
+        $isModule = 'should' === $condition;
+
+        /** @var Carrier $carrier */
+        $carrier = SharedStorage::getStorage()->get($reference);
+        $actualResult = (bool) $carrier->is_module;
+
+        if ($actualResult !== $isModule) {
+            throw new RuntimeException(sprintf(
+                'Carrier "%s" %s to module. It was expected that it %s belong to module',
+                $reference,
+                $actualResult ? 'belongs' : 'does not belong',
+                $condition
+            ));
+        }
+    }
+
+    /**
+     * @Then Carrier :reference name in default language should be :value
+     */
+    public function assertLocalizedName($reference, $value)
+    {
+        /** @var Carrier $carrier */
+        $carrier = SharedStorage::getStorage()->get($reference);
+
+        if (!isset($carrier->localized_name[$this->defaultLangId])) {
+            throw new RuntimeException(sprintf(
+                'Carrier "%s" name in default language is not set, "%s" was expected.',
+                $reference,
+                $value
+            ));
+        }
+        if ($carrier->localized_name[$this->defaultLangId] !== $value) {
+            throw new RuntimeException(sprintf(
+                'Carrier "%s" name in default language is "%s", but "%s" was expected.',
+                $reference,
+                $carrier->localized_name[$this->defaultLangId],
+                $value
+            ));
+        }
+    }
+
+    /**
+     * @Then Carrier :reference shipping delay in default language should be :value
+     */
+    public function assertShippingDelay($reference, $value)
```

Comment:
Do you think you could merge this assertion with `assertLocalizedName()` since they do exactly the same.